File: Homework/HW3/threads_helloWorld.py
#!/usr/bin/python
#***************************************************************
#* 
#* Author: Dr. Lisa Frye
#* Date: September 15, 2015
#* Filename: threads_helloWorld.py
#* Purpose: A simple Hello World program using threads
#*
#**************************************************************

import threading

# thread worker function
# Name: worker
# Arguments: num - an integer
# Return value: none
def worker(num):
    # print number passed to function
    print 'Worker: %s' % num
    return

threads = []   # array to hold each thread ID
# spawn 5 threads
for i in range(5):
    # create the thread
    t = threading.Thread(target=worker, args=(i,))
    threads.append(t)
    t.start()   # start the thread by calling run() method

File: Homework/HW3/threads_helloWorld.c
/******************************************************************************
 *
 * Author: Dr. Lisa Frye
 * Creation Date: August 28, 2015
 * Filename: hello.c
 * Purpose: A Hello World program to illustrate pthreads.
 * Compile:  gcc helloWorld.c -lpthread
 *
******************************************************************************/

#include <stdio.h>
#include <stdlib.h>   // for exit()
#include <pthread.h>  // for pthread functions

#define NUM_THREADS	5

void *print(void *threadid);


int main(int argc, char *argv[])
{
  int i, rc;
  pthread_t threads[NUM_THREADS];
 
  for(i=0; i < NUM_THREADS; i++){
     printf("Creating thread %d\n", i);

     // create a thread and call the print function
     // store ID of new thread in threads array
     // i is passed as the argument to the print function
     rc = pthread_create(&threads[i], NULL, print, (void *)i);

     if (rc){  // if error creating thread
       printf("ERROR; return code from pthread_create() is %d\n", rc);
       exit(-1);
     } // end if
  }  // end for

  pthread_exit(NULL);
}  // end function main


/**************************************************************
 *
 * Function name: print
 * Description: print "Hello World"
 * Parameters: threadid - the ID of the current thread
 * Return Value: none
 *
 **************************************************************/
void *print(void *threadid)
{
   printf("\n%d: Hello World!\n", threadid);

   // terminate the calling hread
   pthread_exit(NULL);
}   // end function print
